### PR TITLE
added unit tests for aliases

### PR DESCRIPTION
**Pull request by tahaalibra**
add unit tests for aliases

### CODE REVIEW

File: tests/controller/aliasescontrollertest.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	private $user;
+
+	public function setUp() {
+		$this->request = $this->getMockBuilder('OCP\IRequest')
+			->getMock();
+		$this->aliasService = $this->getMockBuilder('OCA\Mail\Service\AliasesService')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->userSession = $this->getMockBuilder('OCP\IUserSession')
+			->disableOriginalConstructor()
```
Comment (ChristophWurst) [inline tests/controller/aliasescontrollertest.php]: interfaces do not have a constructor :wink:

File: tests/controller/aliasescontrollertest.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+			->getMock();
+		$this->user = $this->getMockBuilder('OCP\IUser')
+			->disableOriginalConstructor()
```
Comment (ChristophWurst) [inline tests/controller/aliasescontrollertest.php]: same here

File: tests/controller/aliasescontrollertest.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+			->getMock();
+
+		$this->userSession->expects($this->once())
```
Comment (ChristophWurst) [inline tests/controller/aliasescontrollertest.php]: why was this assertion put in to the setUp method? Can we move that down to the individual test methods?

File: tests/controller/aliasescontrollertest.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+use OC\AppFramework\Http;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Http\JSONResponse;
+use OCA\Mail\Controller\AliasesController;
+use OCP\IUserSession;
+
+class AliasesControllerTest extends PHPUnit_Framework_TestCase {
```
Comment (ChristophWurst) [inline tests/controller/aliasescontrollertest.php]: Please extend the core test case instead. See https://github.com/owncloud/mail/blob/afb70cf07c10b5fa22cf287b7480449da9b894b3/tests/applicationtest.php#L24 for an example

File: tests/db/aliasmappertest.php [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+	/**
+	 * @var Alias
+	 */
+	private $alias;
+
+	/**
+	 * Initialize Mapper
+	 */
+	public function setup(){
```
Comment (ChristophWurst) [inline tests/db/aliasmappertest.php]: would be great to have a `tearDown` method too, which restores the previous db state

File: tests/db/aliasmappertest.php [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+		$db = \OC::$server->getDatabaseConnection();
+		$this->db = new Db($db);
+		$this->mapper = new AliasMapper($this->db);
+
+		$this->alias = new Alias();
```
Comment (ChristophWurst) [inline tests/db/aliasmappertest.php]: I'd move this block down, as it's part of the test. Only initialization code should be in the setUp method in my opinion

**Comment by tahaalibra** [comment]
@ChristophWurst can you review please

**Pull request closed without merging**